core/core.go

```diff
@@ -283,8 +323,37 @@ func NewTextile(ctx context.Context, conf Config) (*Textile, error) {
 	}
 	var opts []grpc.ServerOption
 	if conf.Hub {
+		var powStub *grpcdynamic.Stub
+		var healthServiceDesc *desc.ServiceDescriptor
+		var netServiceDesc *desc.ServiceDescriptor
+		var ffsServiceDesc *desc.ServiceDescriptor
+		var walletServiceDesc *desc.ServiceDescriptor
+		if conf.AddrPowergateAPI != "" {
+			if powStub, err = createPowStub(conf.AddrPowergateAPI); err != nil {
+				return nil, err
+			}
+			if healthServiceDesc, err = createServiceDesciptor("health/rpc/rpc.proto", healthServiceName); err != nil {
+				return nil, err
+			}
+			if netServiceDesc, err = createServiceDesciptor("net/rpc/rpc.proto", netServiceName); err != nil {
+				return nil, err
+			}
+			if ffsServiceDesc, err = createServiceDesciptor("ffs/rpc/rpc.proto", ffsServiceName); err != nil {
+				return nil, err
+			}
+			if walletServiceDesc, err = createServiceDesciptor("wallet/rpc/rpc.proto", walletServiceName); err != nil {
+				return nil, err
+			}
+		}
 		opts = []grpc.ServerOption{
-			grpcm.WithUnaryServerChain(auth.UnaryServerInterceptor(t.authFunc), t.threadInterceptor()),
+			grpcm.WithUnaryServerChain(
+				auth.UnaryServerInterceptor(t.authFunc),
+				t.threadInterceptor(),
+				generatePowUnaryInterceptor(healthServiceName, allowedPowMethods[healthServiceName], healthServiceDesc, powStub, t.powc, t.collections),
+				generatePowUnaryInterceptor(netServiceName, allowedPowMethods[netServiceName], netServiceDesc, powStub, t.powc, t.collections),
+				generatePowUnaryInterceptor(ffsServiceName, allowedPowMethods[ffsServiceName], ffsServiceDesc, powStub, t.powc, t.collections),
+				generatePowUnaryInterceptor(walletServiceName, allowedPowMethods[walletServiceName], walletServiceDesc, powStub, t.powc, t.collections),
+			),
```

Comment:
Idea here is to always register the powergate interceptors, even if powergate isn't configured to be running as part of Hub. That way, the interceptors can detect that and return a nice error saying powergate isn't enabled.